content/client-lib-development-guide/features.textile

```diff
@@ -1590,27 +1603,22 @@ enum StatsIntervalGranularity:
 
 class DeviceDetails:
   id: String
-  platform: DevicePlatform
-  formFactor: DeviceFormFactor
   clientId: String?
+  formFactor: DeviceFormFactor
   metadata: JsonObject
-  updateToken: String
+  platform: DevicePlatform
   push: DevicePushDetails
+  updateToken: String
 
 class DevicePushDetails:
-  transportType: DevicePushTransportType
-  transportId: String
-  transportToken: String?
-  state: .Active | .Failing | .Failed
   errorReason: ErrorInfo?
-  metadata: JsonObject
+  recipient: JsonObject
+  state: .Active | .Failing | .Failed
 
 class LocalDevice extends DeviceDetails:
-  resetId()
-  resetUpdateToken() => io
+  resetUpdateToken(token: String?) => io
```

Comment:
`resetUpdateToken` without the parameter is a very different operation that with it (assuming you mean it to set a update token gotten from another request, to handle the 401 case we spoke about). The former will call `resetUpdateToken`, the latter will not. I'd really use two separate methods, don't you think? Otherwise, it may look like passing a token will also try to set it in the server, which isn't possible.